Scripts/RTS/WalkerRunAStar.cs

```diff
@@ -0,0 +1,178 @@
+namespace RTS;
+
+public partial class Walker {
+        [Export]
+        public bool HasDiagonalMovement { get; set; } = true;
+        private AStar2D AStar = new AStar2D();
+        List<Vector2> path = new List<Vector2>();
+        int currentPathIndex = 0;
+        private Vector2 startPoint;
+        private Vector2 endPoint;
+        bool hasRun = false;
+        bool isRunning = false;
+
+
+        public override void _PhysicsProcess(double delta)
+        {
+            base._PhysicsProcess(delta);
+            FollowPath();
+        }
+
+        public void ResetPaths(){
+            
+            startPoint = World.Instance.Trees.GetUsedRect().Position;
+            endPoint = World.Instance.Trees.GetUsedRect().End;
+        }
+
+    // TODO: Bind this to a skill first
+        public void TriggerWalkToPoint(Vector2I Tilecoord)
+        {
+            if(hasRun == false){
+                if(HasDiagonalMovement) AddDiagonalMovement();
```

Comment:
This part should be in something like a _Ready function, but the tilemaps have not loaded in yet if we do place it into the _Ready function. I didn't know if there was a single run lifecycle hook that runs after all _Ready functions have been triggered